src/axom/sidre/core/View.cpp

```diff
@@ -1053,11 +1053,11 @@ void View::deepCopyView(View* copy, int allocID) const
 
   if(isDescribed())
   {
-    copy->describe(m_schema.dtype());
     if(hasBuffer() || m_state == EXTERNAL)
     {
       copy->allocate(getTypeID(), getNumElements(), allocID);
     }
+    copy->describe(getTypeID(), getNumDimensions(), m_shape.data());
```

Comment:
Would having an allocate version that takes a shape remove the need for this correction?  If we had that `allocate` version, we could call it, but what about when case when we don't call `allocate` at all?

Reply:
Does `allocate` call `describe` somehow?  Perhaps that's why the current version calls `describe` first.  @rhornung67 , didn't you have a whole state machine involving describe and allocate, etc?

Reply:
2 of the 3 current `allocate` functions call describe.  I don't really understand the roles of `describe` and `m_schema`.

Reply:
@gunney1 @agcapps is referring to a document I wrote years ago that describes the concepts and how we agreed they would work (after much debate). It's on our CZ confluence space, but it's hard to link here. I will post on our MS Teams space. Caveat: I am not 100% confident that it is consistent with the current state of Axom.

Reply:
The overloaded `allocate` methods which call `describe` call it without a shape, so the View ends up with a default shape (1-dimensional with the data length) and apparently wipes out any previous shape that may have been there.  This may be due to shape being a concept added later in development as a part of the description that's there in addition to the conduit schema.

Reply:
I am adding an `allocate` method that takes a shape, so we don't have to fix the description after allocation.  To answer the question that started this thread: if we don't call `allocate`, then we have to call `describe` to set the description.  There was nothing wrong with messing up the description by calling the other `allocate` then fixing it, but using the right `allocate` makes it clearer.  The next push will have this change.  Thanks everyone for commenting.